forge-game/src/main/java/forge/game/ability/effects/CopySpellAbilityEffect.java

```diff
@@ -97,7 +97,8 @@ public void resolve(SpellAbility sa) {
                 // Find subability or rootability that has targets
                 SpellAbility targetedSA = chosenSA;
                 while (targetedSA != null) {
-                    if (targetedSA.usesTargeting() && targetedSA.getTargets().size() != 0) {
+                    if (targetedSA.usesTargeting() && (targetedSA.getTargets().size() != 0
+                            || (sa.hasParam("Defined") && sa.getParam("Defined").startsWith("Triggered")))) {
```

Comment:
when Forge references targetedSA for Agrus Kos, targetSA.getTargets().size() always returns 0 - hence this hack

not sure this is the best way to solve this though for this card, as BecomesTarget really has done all the checks we need on whether the right thing is targeted

Reply:
What ability did you test? Triggered or Activated?

Reply:
Main test was activated ability of Kabuto Moth

Reply:
You should debug why that is the case, does it have something to do with becomes target trigger?

Reply:
The problem is multiple (from same source) `AbilityActivated` on the stack are usually the same object

What you can try:

Add `AbilityKey.StackInstance` like `TriggerSpellAbilityCastOrCopy` already does
\+ `Defined$ TriggeredStackInstance`

Reply:
if that's the case we really need to cleanup the Core more about this

Reply:
@tool4ever your suggestion seems fine

I think remaining core issues could go to an issue